src/providers/xyz.js

```diff
@@ -0,0 +1,32 @@
+var d3 = require("d3");
+
+function XYZProvider(options) {
+  this.format = options.format;
+  this.tileCache = {};
+  this.urlTemplate = options.urlTemplate;
+}
+
+XYZProvider.prototype = {
+  getTile: function(tilePoint, callback){
+    var tileData = this.tileCache[tilePoint.zoom + ":" + tilePoint.x + ":" + tilePoint.y];
```

Comment:
I added that there provisionally in order to be able to change css properties without reloading tiles. But using d3 properly (in the sense of data updating) will make it unnecessary.